src/irmin/object_graph.ml

```diff
@@ -207,11 +208,11 @@ module Make (Hash : HASH) (Branch : Type.S) = struct
         keys
     in
     let visit key level =
-      if level >= depth then Lwt.return_unit
+      if (* ISSUE *) level >= depth then Lwt.return_unit
       else if has_mark key then Lwt.return_unit
       else
         skip key >>= function
-        | true -> Lwt.return_unit
+        | true -> Lwt.return_unit (* ISSUE *)
```

Comment:
I think we should mark it so that we don't call `skip` several times if that object is reachable from another path.